pysidt/sidt.py

```diff
@@ -657,18 +657,29 @@ def generate_tree(
                 for child in children_to_remove:
                     node.children.remove(child)
 
-            self.fit_tree(data=None, check_data=False, alpha=alpha)
+            self.fit_tree(data=None, check_data=False, alpha=alpha, fit_rule=True, estimate_full_uncertainty=True)
 
-    def fit_tree(self, data=None, check_data=True, alpha=0.1):
+        else:
+
+            self.fit_tree(data=None, check_data=False, alpha=alpha, fit_rule=False, estimate_full_uncertainty=True)
+
+    def fit_tree(self, data=None, check_data=True, alpha=0.1, fit_rule=True, estimate_full_uncertainty=False):
         """
         fit rule for each node
         """
         if data:
             self.setup_data(data, check_data=check_data)
             self.descend_training_from_top(only_specific_match=True)
 
-        max_depth = max([node.depth for node in self.nodes.values()])
+        if fit_rule:
+            self.fit_rule(alpha=alpha)
+        
+        if estimate_full_uncertainty:
```

Comment:
This should be default for now. Perhaps we'll need a faster way later, but I don't think we trust the layer by layer uncertainties yet.